src/passes/OptimizeInstructions.cpp

```diff
@@ -1110,6 +1135,8 @@ struct OptimizeInstructions
       return;
     }
 
+    skipNonNullCast(curr->value);
```

Comment:
Why do we skip nonnull cast here?

Reply:
All the RefAs operations will trap on non null anyhow (including RefAsNonNull, funnily...).